pkg/build/api/types.go

```diff
@@ -266,6 +266,18 @@ const (
 	// secret for pushing the output image.
 	// The build will stay in the pending state until the secret is created, or the build times out.
 	StatusReasonMissingPushSecret = "MissingPushSecret"
+
+	// StatusReasonFailedPostCommitHook indicates that the postcommit failed
+	StatusReasonPostCommitHookFailed = "PostCommitHookFailed"
+
+	// StatusReasonGitCloneFailed
+	StatusReasonGitCloneFailed = "GitCloneFailed"
+
+	// StatusReasonPullImageFailed
+	StatusReasonPullImageFailed = "PullImageFailed"
```

Comment:
@bparees instead of trying to doctrine every possible "reason" as const, we can just make it a voluntary string. How we are going to use these reasons?